### PR TITLE
feat: embed MCP server in HTTP serve command

### LINKED ISSUE #296: As an admin I should be able to run the app as a mcp/rest/web app all at the same time
Right now I can run the application as a mcp server, rest api or web app but I want to be able to run the app once and have all 3 methods of accessing it available at the same time so that I don't have to deliberate execute 3 different apps

### PR DESCRIPTION
**Pull request by akeemphilbert**
## Summary
- Adds `--mcp` flag (default `true`) to `weos serve` that mounts the MCP protocol at `/api/mcp` using Streamable HTTP transport
- Extracts `NewMCPServer()` for reuse across stdio and HTTP transports
- Users can now run REST API, web app, and MCP server from a single `weos serve` command
- MCP endpoint is under the protected API group, inheriting auth/authorization middleware
- `weos mcp` (stdio) remains unchanged for direct LLM piping
- Disable with `--mcp=false` or `MCP_ENABLED=false`

Closes #296

## Test plan
- [ ] `weos serve` starts with MCP enabled at `/api/mcp` by default
- [ ] `weos serve --mcp=false` starts without MCP endpoint
- [ ] `weos mcp` still works on stdio
- [ ] POST MCP initialize request to `http://localhost:8080/api/mcp` returns valid response
- [ ] All existing tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/cli/serve.go [hunk 1]
```diff
@@ -238,6 +246,12 @@ func runServe(cmd *cobra.Command, args []string) error {
 	protected.PUT("/:typeSlug/:id", resourceHandler.Update)
 	protected.DELETE("/:typeSlug/:id", resourceHandler.Delete)
 
+	if serveViper.GetBool("enabled") {
+		mcpHandler := mcpserver.NewHTTPHandler(resourceTypeService, resourceService)
+		e.Any("/mcp", echo.WrapHandler(mcpHandler))
+		fmt.Println("MCP server enabled at /mcp")
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/cli/serve.go]: Fixed — the SPA middleware now skips `/mcp` routes (see `api/middleware/static.go` change in commit ae5452a).

[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/cli/serve.go]: Fixed — MCP handler is now mounted under the `protected` API group at `/api/mcp`, so it inherits the same auth/impersonation/authorization middleware chain as all other API routes. When OAuth is enabled, MCP requests require authentication. See commit 0bd06aa.

File: internal/mcp/handler.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// NewHTTPHandler returns an http.Handler that serves the MCP protocol over
+// Streamable HTTP with all tool groups enabled.
+func NewHTTPHandler(
+	resourceTypeService application.ResourceTypeService,
+	resourceService application.ResourceService,
+) http.Handler {
+	server := NewMCPServer(resourceTypeService, resourceService, nil)
+	return gomcp.NewStreamableHTTPHandler(func(_ *http.Request) *gomcp.Server {
+		return server
+	}, nil)
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/mcp/handler.go]: Added — `handler_test.go` now includes `TestNewHTTPHandler_AcceptsMCPRequest` which mounts the handler via httptest, sends an MCP `initialize` JSON-RPC request, and verifies a valid 200 response containing the server name. Also added tests for handler creation and nil-service error handling. See commit 0bd06aa.

File: internal/cli/serve.go [hunk 3]
```diff
@@ -238,6 +249,20 @@ func runServe(cmd *cobra.Command, args []string) error {
...
 
+	if serveViper.GetBool("enabled") {
+		mcpHandler, mcpErr := mcpserver.NewHTTPHandler(
+			resourceTypeService, resourceService, slog.Default(),
+		)
+		if mcpErr != nil {
+			return fmt.Errorf("failed to create MCP handler: %w", mcpErr)
+		}
+		protected.Any("/mcp", echo.WrapHandler(mcpHandler))
+		protected.Any("/mcp/*", echo.WrapHandler(mcpHandler))
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/cli/serve.go:247]: This was already addressed in commit 0bd06aa — MCP is now mounted under the `protected` API group at `/api/mcp`, inheriting the same auth/impersonation/authorization middleware as all other API routes. This comment appears to be on the stale `/mcp` route which no longer exists.

File: internal/mcp/server.go [hunk 4]
```diff
@@ -128,10 +120,40 @@ func Run(enabledServices []string) error {
 		registerResourceTools(server, resourceService)
 	}
 
+	return server, nil
+}
+
+// Run starts the MCP server on stdio, registering only the tool groups listed in enabledServices.
+// If enabledServices is empty, all tool groups are registered.
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/mcp/server.go]: Fixed — updated the `Run()` doc comment to say "nil or empty". See commit f6f1052.

File: internal/cli/serve.go [hunk 5]
```diff
@@ -53,6 +58,12 @@ var serveCmd = &cobra.Command{
 }
 
 func init() {
+	serveCmd.Flags().Bool("mcp", true, "enable MCP server over HTTP at /mcp")
+	serveViper.SetEnvPrefix("MCP")
+	serveViper.AutomaticEnv()
+	if err := serveViper.BindPFlag("enabled", serveCmd.Flags().Lookup("mcp")); err != nil {
+		panic(fmt.Sprintf("failed to bind MCP flag: %v", err))
+	}
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/cli/serve.go]: Fixed — flag help text now reads "/api/mcp". See commit 2b0ce3e.

File: internal/cli/serve.go [hunk 6]
```diff
@@ -238,6 +249,20 @@ func runServe(cmd *cobra.Command, args []string) error {
...
+			resourceTypeService, resourceService, slog.Default(),
+		)
+		if mcpErr != nil {
+			return fmt.Errorf("failed to create MCP handler: %w", mcpErr)
+		}
+		protected.Any("/mcp", echo.WrapHandler(mcpHandler))
+		protected.Any("/mcp/*", echo.WrapHandler(mcpHandler))
+		logger.Info(context.Background(), "MCP server enabled", "path", "/api/mcp")
+	} else {
+		logger.Info(context.Background(), "MCP server disabled via configuration")
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/cli/serve.go:250]: Fixed — updating PR description to reference `/api/mcp`. The handler is mounted under the protected API group.

File: internal/mcp/server.go [hunk 7]
```diff
@@ -84,26 +84,18 @@ func resolveEnabled(services []string) map[ServiceName]bool {
...
+	resourceTypeService application.ResourceTypeService,
+	resourceService application.ResourceService,
+	enabledServices []string,
+) (*mcp.Server, error) {
+	if resourceTypeService == nil {
+		return nil, fmt.Errorf("resourceTypeService must not be nil")
+	}
+	if resourceService == nil {
+		return nil, fmt.Errorf("resourceService must not be nil")
 	}
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/mcp/server.go]: Fixed — added `isNilInterface()` helper using `reflect` to catch typed-nil values. See commit 2b0ce3e.

File: internal/mcp/handler_test.go [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+	serverTransport, clientTransport := gomcp.NewInMemoryTransports()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	done := make(chan error, 1)
+	go func() {
+		_, err := server.Connect(ctx, serverTransport, nil)
+		done <- err
+	}()
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/mcp/handler_test.go]: Fixed — test helper now cancels the context and waits on the server goroutine channel, failing the test if `server.Connect` returns an unexpected error. See commit 2b0ce3e.

File: internal/mcp/server.go [hunk 9]
```diff
@@ -128,10 +121,40 @@ func Run(enabledServices []string) error {
...
+	defer startCancel()
+
+	if err := app.Start(startCtx); err != nil {
+		return fmt.Errorf("failed to start application: %w", err)
+	}
+
+	server, err := NewMCPServer(resourceTypeService, resourceService, enabledServices)
+	if err != nil {
+		return fmt.Errorf("failed to create MCP server: %w", err)
+	}
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/mcp/server.go:165]: Fixed — `Run()` now uses `defer` to ensure `app.Stop()` is always called after a successful `app.Start()`, even if `NewMCPServer` returns an error. See commit a743efb.

File: internal/cli/serve.go [hunk 10]
```diff
@@ -238,6 +249,20 @@ func runServe(cmd *cobra.Command, args []string) error {
...
+		mcpHandler, mcpErr := mcpserver.NewHTTPHandler(
+			resourceTypeService, resourceService, slog.Default(),
+		)
+		if mcpErr != nil {
+			return fmt.Errorf("failed to create MCP handler: %w", mcpErr)
+		}
+		protected.Any("/mcp", echo.WrapHandler(mcpHandler))
+		protected.Any("/mcp/*", echo.WrapHandler(mcpHandler))
+		logger.Info(context.Background(), "MCP server enabled", "path", "/api/mcp")
+	} else {
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/cli/serve.go]: Fixed — MCP routes are now registered before the dynamic catch-all `/:typeSlug` routes, consistent with the existing pattern. See commit 978d082.

File: internal/mcp/server.go [hunk 11]
```diff
@@ -84,26 +85,18 @@ func resolveEnabled(services []string) map[ServiceName]bool {
...
-
-	if err := app.Start(startCtx); err != nil {
-		return fmt.Errorf("failed to start application: %w", err)
+// NewMCPServer creates a configured MCP server with the specified tool groups registered.
+// If enabledServices is nil or empty, all tool groups are registered.
+func NewMCPServer(
+	resourceTypeService application.ResourceTypeService,
+	resourceService application.ResourceService,
+	enabledServices []string,
+) (*mcp.Server, error) {
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply internal/mcp/server.go:95]: Fixed — `NewMCPServer` now calls `ValidateServiceNames()` when `enabledServices` is non-empty, returning an error for unknown service names. See commit c52cfbd.
